### PR TITLE
[Maint] Fix comprehensive tests by skipping labels controls test on py311 pyqt6

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
# Fixes/Closes

Closes # https://github.com/napari/napari/issues/5921

# Description

In #5439 @Carreau identified a failing condition for the qt_layer_controls test that a results in a fault on python 3.11 and pyqt5 on ubuntu. However, in that PR, pyqt6 was not tested with python 311. The same test can be see to fail in comprehensive tests which do run python 311 and pyqt6.
This PR adds pyqt6 to the check to skip the test.

# References

#5439 

## Type of change

- [x] CI test fix

# How has this been tested?

### CODE REVIEW

**Review by Carreau — APPROVED** [review]
+1, I think the other test failures are unrelated.

**Comment by psobolewskiPhD** [comment]
@Czaki I see that https://github.com/napari/napari/pull/5439 is 0.4.18 so this should be as well, even though it just touches CI?

File: napari/_qt/layer_controls/_tests/test_qt_layer_controls.py [hunk 1]
```diff
@@ -178,11 +178,16 @@ def test_create_layer_controls(
             qcombobox.setCurrentIndex(qcombobox_initial_idx)
 
 
-if sys.version_info[:2] == (3, 11) and qtpy.API == 'pyqt5':
+if (
+    sys.version_info[:2] == (3, 11)
+    and qtpy.API == 'pyqt5'
+    or qtpy.API == 'pyqt6'
```
Comment (Czaki) [inline napari/_qt/layer_controls/_tests/test_qt_layer_controls.py]:
```suggestion
    sys.version_info[:2] == (3, 11)
    and (qtpy.API == 'pyqt5' or qtpy.API == 'pyqt6')
```

as 
```python
In [1]: False and False or True
Out[1]: True

In [2]: False and (False or True)
Out[2]: False
```

So this clause disables it on any pyqt6 test, not only python 3.11.

Comment (psobolewskiPhD) [reply napari/_qt/layer_controls/_tests/test_qt_layer_controls.py]: Christ, sorry about that. 😭

Comment (Czaki) [reply napari/_qt/layer_controls/_tests/test_qt_layer_controls.py]: Operator precedence is not simple and it is easy to do some mistake.

**Review by Czaki — APPROVED** [review]

**Comment by Carreau** [comment]
As a note, can we please avoid including github user names in descriptions to avoid having them in commit messages. When a name is in a commit message (in that case me), I get notification each time the commit is pushed into a new branch, which may include forks or people playing with the repository.

**Comment by Czaki** [comment]
I'm sorry for this. (As I'm the caller) and 100% agree, as I also got multiple notifications.

**Comment by psobolewskiPhD** [comment]
> As a note, can we please avoid including github user names in descriptions to avoid having them in commit messages. When a name is in a commit message (in that case me), I get notification each time the commit is pushed into a new branch, which may include forks or people playing with the repository.

That's totally my bad—I had no idea that would/could happen! 😬 
Sorry!

(It explains some puzzling notifications I've also gotten recently, so I guess the more you know...)

**Pull request merged**